examples/worker-process-jobs.js

```diff
@@ -0,0 +1,59 @@
+var close, closeIfCan, kue, queue;
+
+kue = require('../');
+
+queue = kue.createQueue({
+  disableSearch: true
+});
+
+close = function() {
+   queue.shutdown(1000, function(err) {
+    if (err) {
+      console.log(err);
+    }
+    console.log('queue closed');
+  });
+};
+
+closeIfCan = function() {
+  queue.inactiveCount(function(err, inactive) {
+    queue.activeCount(function(err, active) {
+      queue.delayedCount(function(err, delayed) {
+        console.log('inactive:', inactive);
+        console.log('active:', active);
+        console.log('delayed:', delayed);
+        if (inactive + active === 0) {
+          close();
+        }
+      });
+    });
+  });
+};
+
+queue.on('job complete', function(result) {
+  console.log('Job completed with result', result);
+  closeIfCan();
+}).on('promotion:end', function() {
```

Comment:
You cannot depend on `promotion:end` being called in every node in a clustered deployment.
Should we add a mock `promotion:end` event to all nodes that are silent (losers to lock race)? or re-view the whole stuff again?